go.mod

```diff
@@ -2,6 +2,8 @@ module go.viam.com/rdk
 
 go 1.21
 
+toolchain go1.21.7
```

Comment:
revert before merge.

Reply:
Why have this?

Reply:
go 1.21 and further uses toolchains to manage the dependencies, running make lint-go sometimes adds this I've noticed. Abe is trying to figure out how to handle adding a toolchain to the go.mod file. Until then I'll remove the toolchain when it's added.